catalog/cataloger_cache.go

```diff
@@ -1,37 +1,49 @@
 package catalog
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/treeverse/lakefs/cache"
 	"github.com/treeverse/lakefs/db"
 )
 
 func (c *cataloger) getRepositoryCache(tx db.Tx, repository string) (*Repository, error) {
-	return c.cache.Repository(repository, func(repository string) (*Repository, error) {
+	repo, err := c.cache.Repository(repository, func(repository string) (*Repository, error) {
 		repo, err := getRepository(tx, repository)
 		if err != nil {
 			return nil, fmt.Errorf("get repository: %w", err)
 		}
 		return repo, nil
 	})
+	return repo, catalogerTransformCacheError(err)
 }
 
 func (c *cataloger) getRepositoryIDCache(tx db.Tx, repository string) (int, error) {
-	return c.cache.RepositoryID(repository, func(repository string) (int, error) {
+	repoID, err := c.cache.RepositoryID(repository, func(repository string) (int, error) {
 		repoID, err := getRepositoryID(tx, repository)
 		if err != nil {
 			return 0, fmt.Errorf("get repository id: %w", err)
 		}
 		return repoID, nil
 	})
+	return repoID, catalogerTransformCacheError(err)
 }
 
 func (c *cataloger) getBranchIDCache(tx db.Tx, repository string, branch string) (int64, error) {
-	return c.cache.BranchID(repository, branch, func(repository string, branch string) (int64, error) {
+	branchID, err := c.cache.BranchID(repository, branch, func(repository string, branch string) (int64, error) {
 		branchID, err := getBranchID(tx, repository, branch, LockTypeNone)
 		if err != nil {
 			return 0, fmt.Errorf("get branch id: %w", err)
 		}
 		return branchID, nil
 	})
+	return branchID, catalogerTransformCacheError(err)
+}
+
+func catalogerTransformCacheError(err error) error {
```

Comment:
1. Couldn't we just fold it into the cache?
1. It transforms one error into a less informative form.  Maybe just rename `ErrCacheItemNotFound` to `ErrNotFound` _in the cache_, and then it makes sense and doesn't say "DB"?

Reply:
The issue is that the cache system doesn't know the operation (get) that fetch the actual data. In this case we fail to find an item from the database, the code above expect to get a DB error saying that the entry not found.  The cache system (generic) hide this information.
At first I thought of using the last fetch/get error as the cache return value, but the the cache uses a mechanism not to perform DB operation why other is pending, not to load DB calls in parallel. So we don't have the actual error. Keeping the last error in the cache for 'not found' is also something I prefer to avoid.
Because we have layer that wraps the cache which implements the fetch/get method it can convert the error to not found.
In case the database error is something else - only the one that perform the actual get/fetch will report the error.

Reply:
I mean the exact opposite: why are we generating `db.ErrNotFound` here?  It is very clearly *not* a detailed DB error, it is *exactly* a "cache item not found error".  Why not return such an error?